src/index.ts

```diff
@@ -327,6 +287,21 @@ export function tuple (...decoders: Decoder<any>[]): Decoder<any> {
   ))
 }
 
+export function field <T>(key: string, decoder: Decoder<T>): Decoder<T> {
+  return createDecoder((obj, at) => {
+    const value = obj[key];
+    if (value === undefined) {
```

Comment:
#10 allows `undefined` to be passed into value decoders, so this will have to change. The new approach should be to simply pass `undefined` to the decoder, and then if it fails, check for the presence of the key and then throw the `missing key` error.

Take a look at [this bit of code](https://github.com/ooesili/type-safe-json-decoder/blob/master/src/index.ts#L260-L268) to get a better idea of what I'm talking about.